Sources/SwiftParserDiagnostics/ParserDiagnosticMessages.swift

```diff
@@ -330,6 +333,34 @@ public struct MissingAttributeArgument: ParserError {
   }
 }
 
+public struct MissingConditionInStatement: ParserError {
+    let firstToken: TokenSyntax
+
+    public var message: String {
+        if let name = firstToken.parent?.ancestorOrSelf(mapping: {
+            $0.nodeTypeNameForDiagnostics(allowBlockNames: false)
+        }) {
+            return "missing condition in \(name)"
+        } else {
+            return "missing condition in statement"
+        }
+    }
+}
+
+public struct MissingExpressionInStatement: ParserError {
+    let firstToken: TokenSyntax
+
+    public var message: String {
+        if let name = firstToken.parent?.ancestorOrSelf(mapping: {
+            $0.nodeTypeNameForDiagnostics(allowBlockNames: false)
+        }) {
```

Comment:
Can't you just pass in the node that contains the `nodeTypeNameForDiagnostics` here (i.e. `IfExpr`, `SwitchExpr` etc.) instead of having to rely on `ancestorOrSelf`?